seqr/views/apis/dashboard_api_tests.py

```diff
@@ -120,3 +122,116 @@ def test_export_projects_table(self):
         response = self.client.get('{}?file_format=csv'.format(url))
         self.assertEqual(response.status_code, 400)
         self.assertDictEqual(response.json(), {'error': 'Invalid file_format: csv'})
+
+
+# Tests for AnVIL access disabled
+class LocalDashboardPageTest(AuthenticationTestCase, DashboardPageTest):
+    fixtures = ['users', '1kg_project']
+
+
+# Test for permissions from AnVIL only
+@mock.patch('seqr.views.utils.terra_api_utils.TERRA_API_ROOT_URL', TEST_TERRA_API_ROOT_URL)
+class AnvilDashboardPageTest(AnvilAuthenticationTestCase, DashboardPageTest):
+    fixtures = ['users', 'social_auth_data', '1kg_project']
+
+    def setUp(self):
+        patcher = mock.patch('seqr.views.utils.permissions_utils.list_anvil_workspaces')
+        patcher.start().side_effect = get_workspaces_side_effect
+        self.addCleanup(patcher.stop)
+        patcher = mock.patch('seqr.views.utils.terra_api_utils._service_account_session')
+        patcher.start().get.side_effect = get_ws_acl_side_effect
+        self.addCleanup(patcher.stop)
+        patcher = mock.patch('seqr.views.utils.terra_api_utils.time')
+        patcher.start().return_value = 1603287741 + 10
+        self.addCleanup(patcher.stop)
+
```

Comment:
you need to test that the dashboard page is calling all the mocked methods correctly. In the setup method, you can attach the patchers to the class so you can access them in the test:
```
patcher = mock.patch('seqr.views.utils.permissions_utils.list_anvil_workspaces')
self.mock_list_workspaces = patcher.start()
self.mock_list_workspaces.side_effect = get_workspaces_side_effect
```
Then you can use them the way you do mocks that were added by decorator:
```
def test_dashboard_page_data(self):
   super(AnvilDashboardPageTest , self).test_dashboard_page_data()
   self.mock_list_workspaces.assert_has_calls(...)
```

Reply:
Okay. But the pure anvil tests won't check the mocking because it re-uses the existing tests without changes. It must be safe because the local permissions are disabled so that the permissions can only come from the mocking.